File: Src/Lives2D_Main/Lives2D.cpp
#include"Lives2D.h"
#include"Tools\Helper.h"
#include"glm\glm.hpp"
#include"KeyTouch\KeyTouch.h"


int Lives2D::m_Width = 0;
int Lives2D::m_Height = 0;

float Lives2D::m_DesignWidth = 960.0f;
float Lives2D::m_DesignHeight = 540.0f;

DemoScene* Lives2D::mDemoScene = NULL;


void Lives2D::Init(EGLSurface & varEGLSurface, EGLDisplay & varEGLDisplay, int & varWidth, int & varHeight)
{
	m_Width = varWidth;
	m_Height = varHeight;


	mDemoScene = new DemoScene();
	mDemoScene->Init();
	
}



void Lives2D::Update(float varDeltaTime)
{
	mDemoScene->Update(varDeltaTime);
}



void Lives2D::Draw()
{
	mDemoScene->Draw();
}

void Lives2D::OnKey(KeyCode varkeyCode)
{
	KeyTouch::OnKey(varkeyCode);
}

void Lives2D::OnTouch(int varX, int varY)
{
	KeyTouch::OnTouch(varX, varY);
}

void Lives2D::OnTouchRelease(int varX, int varY)
{
	KeyTouch::OnTouchRelease(varX, varY);
}

File: Resources/Script/Engine/Lives2D.lua
local GameMain=require("GameMain")

print("Lives2D")

--初始化
function Init(varWidth,varHeight)
	print("Lives2D Init " .. varWidth .. varHeight)

	AudioCardInit()

	GameMain:Init()
end

--刷帧
function Update(varDeltaTime)
	--print("Update " .. varDeltaTime)
	GameMain:Update(varDeltaTime)
end

--绘制
function Draw()
	--print("Draw")
	GameMain:Draw()
end

--键盘鼠标
function OnKey(varkeyCode)
	--print("OnKey " .. varkeyCode)
	
	KeyTouch:OnKey(varkeyCode)
end

--屏幕触摸点击
function OnTouch(varX,varY)

	--print("OnTouch " .. varX .." " .. varY)
	KeyTouch:OnTouch(varX, varY)
end


--释放点击
function OnTouchRelease(varX,varY)
	--print("OnTouchRelease " .. varX .." " .. varY)

	KeyTouch:OnTouchRelease(varX, varY)
end

function OnDestroy()
	print("OnDestroy")
	AudioCardExit()
end